src/Common/src/TypeSystem/IL/ILImporter.cs

```diff
@@ -116,6 +116,33 @@ private BasicBlock CreateBasicBlock(int offset)
                 basicBlock = new BasicBlock() { StartOffset = offset };
                 _basicBlocks[offset] = basicBlock;
             }
+
```

Comment:
The ECMA-335 requires ordering of the nested try blocks, but not ordering of the non-nested try blocks unfortunately. It is described in detail at: https://github.com/dotnet/coreclr/blob/master/src/jit/jiteh.cpp#L1814

I do not think you can use the simple for-loop to find the handler index because of this - it would only work if the array is sorted first.